src/app/modules/form/admin-form/__tests__/admin-form.service.spec.ts

```diff
@@ -2242,4 +2246,106 @@ describe('admin-form.service', () => {
       expect(expected._unsafeUnwrapErr()).toBeInstanceOf(DatabaseError)
     })
   })
+
+  describe('shouldUpdateFormField', () => {
+    const MOCK_FORM = {
+      admin: {
+        _id: new ObjectId(),
+      },
+    } as unknown as IPopulatedForm
+
+    const MOCK_UNVERIFIABLE_MOBILE_FIELD = generateDefaultField(
+      BasicField.Mobile,
+    )
+    const MOCK_VERIFIABLE_MOBILE_FIELD = generateDefaultField(
+      BasicField.Mobile,
+      { isVerifiable: true },
+    )
+
+    const countSpy = jest.spyOn(SmsService, 'retrieveFreeSmsCounts')
+
+    it('should return the form when admin is under the limit', async () => {
```

Comment:
should we increase the specificity of this test description? we are after all in AdminFormService, and it's not obvious what this is checking.

suggestion "should return the given form when form field to update is a mobile field and admin is still under the free SMS quota limit"

Reply:
we can even have a deeper describe block of 
```ts
describe('when update form field is of BasicField.Mobile type', () => {
  ...
})
```

Reply:
i think having a deeper `describe` block makes sense because it's a special case so i'll adjust it accordingly!